Docs: Data links preserve ad hoc filter and variable context

sidharth-chauhan opened the pull request:
### Summary
Adds a short Tip to the **Add variables → Ad hoc filters** page explaining that data links carry the current time range, dashboard variables, and ad hoc filters so the destination opens with the same filters applied. The Tip also clarifies that ad hoc filters are data-source scoped and only apply when the destination uses the same data source.

### Files changed
- `docs/sources/dashboards/variables/add-template-variables/index.md`

### Why
This small docs addition improves discoverability and prevents confusion when users follow data links between dashboards or panels. It resolves the gap reported in grafana/grafana#110915.

### How I tested
- Built and previewed docs locally (`make run` in `docs/`) and verified the Tip renders under **Add ad hoc filters**.

Closes: grafana/grafana#110915
<img width="1135" height="853" alt="Screenshot 2025-09-11 at 10 21 09 PM" src="https://github.com/user-attachments/assets/c0cf1f4e-91c0-4feb-9c30-de85e45d082b" />

sidharth-chauhan committed:
docs: add provisioning endpoints proposal (draft)

sidharth-chauhan committed:
docs: note that data links preserve ad hoc filter and variable context

sidharth-chauhan committed:
docs: note that data links preserve ad hoc filter and variable context

sidharth-chauhan committed:
docs: note that data links preserve ad hoc filter and variable context

sidharth-chauhan committed:
Merge remote-tracking branch 'upstream/main' into docs/data-links-adhoc-context

sidharth-chauhan committed:
docs: note that data links preserve ad hoc filters & variable context

samjewell commented:
Sorry for my slow response here. It's hackathon week at Grafana, and we're trying to get some things working. I hope I can get to this next week though.

samjewell commented:
https://github.com/user-attachments/assets/4ffc9598-b745-48e5-8f2c-8488afe84d2c

Hi @sidharth-chauhan - sorry if my second video didn't work nicely. I've just compressed it further, see if this works.

samjewell reviewed (changes requested):
See my previous comments

sidharth-chauhan committed:
Clarify data links and ad hoc filters usage

Updated the section on data links to clarify how to preserve context when linking back to the current dashboard and included details on ad hoc filters.

samjewell reviewed (approved):
Awesome, I love it. Thanks a lot for this, it's really great now.

Thanks also for taking my feedback on board, and having another go at this 👏

imatwawana committed:
Updated URLs

imatwawana committed:
Merge branch 'main' into docs/data-links-adhoc-context

imatwawana committed:
Fixed formatting

imatwawana committed:
Fixed formatting and ran prettier
